### DISCUSSION TITLE
Start language server for .langium grammar files

### DISCUSSION THREAD

**Discussion by borisbrodski** [eclipse-langium/langium#1588] [Q&A] [answered]
I would like to use NeoVim for development. How to manually run language server for .langium grammar files?

(Should be straight forward, but I couldn't find it even after extensive web search)

**Comment by msujew** [comment, marked as answer]
Hey @borisbrodski,

you would need to extract the [`main.js`](https://github.com/eclipse-langium/langium/blob/main/packages/langium-vscode/src/language-server/main.ts) from the vscode extension (or build it from scratch) and run the file using `node main.js --stdio` and connect the stdio with the language client. No idea how that works in NeoVim though. We just use the `vscode-languageclient` to get the support for VS Code.

Also, does NeoVim support textmate based syntax highlighting? We make heavy use of that to get the syntax highlighting in VS Code.

**Comment by borisbrodski** [comment]
This works indeed. Thank you very much!

I just needed to fix one line in the code: https://github.com/eclipse-langium/langium/pull/1590
